source/shared/cpp/ObjectModel/MarkDownParsedResult.cpp

```diff
@@ -0,0 +1,227 @@
+#include "MarkDownParsedResult.h"
+
+using namespace AdaptiveCards;
+
+void MarkDownParsedResult::Translate()
+{
+    MatchLeftAndRightEmphasises();
+}
+
+// appends html block tags at head and tail of the list
+void MarkDownParsedResult::AddBlockTags()
+{
+    // if html list tag, add <ul>, otherwise <p> at the head of the list
+    std::string tags = (m_codeGenTokens.front()->IsList())? "<ul>" : "<p>";
```

Comment:
>"<ul>" : "<p>"; [](start = 60, length = 15)

Generating specific tags feels like it belongs in the HtmlGenerator code... Is it possible to make this happen as part of the GenerateHtmlString for MarkDownListHtmlGenerator?

Reply:
no, ListHtmlGenerator can't tell if the previous html generator is <p> type or <li> list type.

---
In reply to: [157628776](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157628776) [](ancestors = 157628776)

Reply:
i moved code gen part to GenerateHtmlString

---
In reply to: [157629389](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157629389) [](ancestors = 157629389,157628776)